base/dates/io.jl

```diff
@@ -324,22 +433,41 @@ generate the string "1996-01-15T00:00:00" you could use `format`: "yyyy-mm-ddTHH
 Note that if you need to use a code character as a literal you can use the escape character
 backslash. The string "1996y01m" can be produced with the format "yyyy\\ymm\\m".
 """
-format(dt::TimeType,f::AbstractString;locale::AbstractString="english") = format(dt,DateFormat(f,locale))
+format(dt::TimeType,f::AbstractString;locale=ENGLISH) = format(dt,DateFormat(f,locale))
+
+# show
+
+function Base.show(io::IO, dt::DateTime)
+    df = if millisecond(dt) == 0
+        dateformat"YYYY-mm-dd\THH:MM:SS"
+    else
+        dateformat"YYYY-mm-dd\THH:MM:SS.s"
+    end
```

Comment:
Glad it's not a problem. If we were to change this I think I would prefer `"YYYY-mm-dd\THH:MM:SS.sss"`